table/tables/index.go

```diff
@@ -95,9 +97,20 @@ func (c *indexIter) Next() (val []types.Datum, h int64, err error) {
 
 // index is the data structure for index data in the KV store.
 type index struct {
-	idxInfo *model.IndexInfo
-	tblInfo *model.TableInfo
-	prefix  kv.Key
+	idxInfo       *model.IndexInfo
+	tblInfo       *model.TableInfo
+	prefix        kv.Key
+	containString bool
+}
+
+func (c *index) checkContainNonBinaryString() bool {
+	for _, idxCol := range c.idxInfo.Columns {
+		col := c.tblInfo.Columns[idxCol.Offset]
+		if types.EvalType(col.Tp) == types.ETString && !mysql.HasBinaryFlag(col.Flag) {
```

Comment:
```suggestion
		if col.EvalType() == types.ETString && !mysql.HasBinaryFlag(col.Flag) {
```
`types.EvalType` is a type definition rather than a function.